### PR TITLE
Convert Event to use factory map

### PR DESCRIPTION
**Pull request by wthrowe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: src/Elliptic/Executables/Xcts/SolveXcts.hpp [hunk 1]
```diff
@@ -145,36 +146,35 @@ struct Metavariables {
...
       : tt::ConformsTo<Options::protocols::FactoryCreation> {
-    using factory_classes = tmpl::map<tmpl::pair<
-        Trigger, tmpl::push_back<Triggers::logical_triggers,
-                                 elliptic::Triggers::EveryNIterations<
-                                     nonlinear_solver_iteration_id>>>>;
+    using factory_classes = tmpl::map<
+        tmpl::pair<Event, tmpl::flatten<tmpl::list<
+                              Events::Completion,
+                              dg::Events::field_observations<
+                                  volume_dim, linear_solver_iteration_id,
```
Comment (nilsvu) [inline src/Elliptic/Executables/Xcts/SolveXcts.hpp]: `linear_solver_iteration_id` -> `nonlinear_solver_iteration_id`

File: src/ParallelAlgorithms/EventsAndTriggers/CMakeLists.txt [hunk 2]
```diff
@@ -34,4 +35,9 @@ target_link_libraries(
   Utilities
   )
 
+add_dependencies(
+  ${LIBRARY}
+  module_GlobalCache
+  )
+
```
Comment (nilsvu) [inline src/ParallelAlgorithms/EventsAndTriggers/CMakeLists.txt]: This dependency should be added to `Parallel` instead

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: src/Parallel/CMakeLists.txt [hunk 3]
```diff
@@ -64,6 +64,7 @@ target_link_libraries(
 
 add_dependencies(
   ${LIBRARY}
+  module_GlobalCache
```
Comment (nilsvu) [inline src/Parallel/CMakeLists.txt]: (optional) this could reasonably be a separate commit

**Review by nilsvu — DISMISSED** [review]

File: src/Parallel/CMakeLists.txt (hunk 3 above)
Comment (nilsvu) [reply src/Parallel/CMakeLists.txt]: Oh actually you just have to rebase, I added this already in #3144

**Review by nilsdeppe — APPROVED** [review]

**Review by nilsvu — APPROVED** [review]

**Pull request merged**